src/NServiceBus.NHibernate/SynchronizedStorage/NHibernateAmbientTransactionSynchronizedStorageSession.cs

```diff
@@ -23,14 +23,20 @@ public NHibernateLazyAmbientTransactionSynchronizedStorageSession(Func<IDbConnec
 
         public void Dispose()
         {
-            if (session.IsValueCreated)
+            try
```

Comment:
Looks like connection will never be created if session has not been accessed, so check on session should be enough. You have a good eye @SimonCropp.

I'd leave these more conservative checks in hotfixes and go with a single `if` on develop. Thoughts?